routers/api/v1/repo/pull.go

```diff
@@ -187,6 +187,91 @@ func GetPullRequest(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, convert.ToAPIPullRequest(ctx, pr, ctx.Doer))
 }
 
+// GetPullRequest returns a single PR based on index
+func GetPullRequestByBaseHead(ctx *context.APIContext) {
```

Comment:
a simple test to https://github.com/go-gitea/gitea/blob/main/tests/integration/api_pull_test.go would be nice,

one for 404
one for local pull
one for pull from fork (if fixtures allow it)